music/src/core/sequences.ts

```diff
@@ -325,3 +334,58 @@ export function assertIsAbsoluteQuantizedSequence(ns: INoteSequence) {
         ns.id} is not quantized or is quantized based on relative timing`);
   }
 }
+
+/**
+ * Create an unquantized version of a quantized `NoteSequence`.
+ *
+ * Any existing times will be replaced in the output `NoteSequence` and
+ * quantization info and steps will be removed.
+ *
+ * @param ns The `NoteSequence` to unquantize.
+ * @param qpm The tempo to use. If not provided, the tempo in `ns` is used,
+ * or the default of 120 if it is not specified in the sequence either.
+ * @returns a new non-quantized `NoteSequence` wih time in seconds.
+ */
+export function unquantizeSequence(qns: INoteSequence, qpm?: number) {
+  // TODO(adarob): Support absolute quantized times and multiple tempos.
+  assertIsRelativeQuantizedSequence(qns);
+  assertSingleTempo(qns);
+
+  const ns = clone(qns);
+
+  if (qpm) {
+    if (ns.tempos && ns.tempos.length > 0) {
+      ns.tempos[0].qpm = qpm;
+    } else {
+      ns.tempos.push(NoteSequence.Tempo.create({time: 0, qpm: qpm}));
+    }
+  } else {
+    qpm = (qns.tempos && qns.tempos.length > 0) ?
+        ns.tempos[0].qpm :
+        constants.DEFAULT_QUARTERS_PER_MINUTE;
+  }
+
+  const stepToSeconds = (step: number) =>
+      step / ns.quantizationInfo.stepsPerQuarter * (60 / qpm);
+  ns.totalTime = stepToSeconds(ns.totalQuantizedSteps);
+  ns.totalQuantizedSteps = undefined;
+  ns.notes.forEach(n => {
+    // Quantize the start and end times of the note.
+    n.startTime = stepToSeconds(n.quantizedStartStep);
+    n.endTime = stepToSeconds(n.quantizedEndStep);
+    n.quantizedStartStep = undefined;
+    n.quantizedEndStep = undefined;
+
+    // Extend sequence if necessary.
+    ns.totalTime = Math.max(ns.totalTime, n.endTime);
+  });
+
+  // Also quantize control changes and text annotations.
+  ns.controlChanges.concat(ns.textAnnotations).forEach(event => {
```

Comment:
for completeness: time signatures, key signatures, pitch bends, section annotations. tempos too, but those are taken care of above.

it might be useful to have a function that returns a concatenated list of these timed events for any given sequence.

Reply:
done for those event types that have quantized steps